File: gex/Models/MapStats/MapStatsOpeningLab.cs
using Dapper.ColumnMapper;
using gex.Code;
using gex.Code.Constants;
using System;

namespace gex.Models.MapStats {

    [DapperColumnsMapped]
    public class MapStatsOpeningLab {

        [ColumnMapping("map_file_name")]
        public string MapFilename { get; set; } = "";

        [ColumnMapping("map_file_name")]
        public byte Gamemode { get; set; } = BarGamemode.DEFAULT;

        [ColumnMapping("def_name")]
        public string DefName { get; set; } = "";

        [ColumnMapping("timestamp")]
        public DateTime Timestamp { get; set; }

        [ColumnMapping("count_total")]
        public int CountTotal { get; set; }

        [ColumnMapping("win_total")]
        public int WinTotal { get; set; }

        [ColumnMapping("count_month")]
        public int CountMonth { get; set; }

        [ColumnMapping("win_month")]
        public int WinMonth { get; set; }

        [ColumnMapping("count_week")]
        public int CountWeek { get; set; }

        [ColumnMapping("win_week")]
        public int WinWeek { get; set; }

        [ColumnMapping("count_day")]
        public int CountDay { get; set; }

        [ColumnMapping("win_day")]
        public int WinDay { get; set; }

    }
}

File: gex/src/model/map_stats/MapStatsOpeningLab.ts

export class MapStatsOpeningLab {
    public mapFilename: string = "";
    public gamemode: number = 0;
    public defName: string = "";
    public timestamp: Date = new Date();
    public countTotal: number = 0;
    public winTotal: number = 0;
    public countMonth: number = 0;
    public winMonth: number = 0;
    public countWeek: number = 0;
    public winWeek: number = 0;
    public countDay: number = 0;
    public winDay: number = 0;

    public static parse(elem: any): MapStatsOpeningLab {
        return {
            ...elem,
            timestamp: new Date(elem.timestamp)
        };
    }
}